src/pybind/mgr/snap_schedule/fs/schedule_client.py

```diff
@@ -180,6 +180,41 @@ def allow_minute_snaps(self) -> None:
     def dump_on_update(self) -> None:
         return self.mgr.get_module_option('dump_on_update')
 
+    def _create_snap_schedule_kv_db(self, db: sqlite3.Connection) -> None:
+        SQL = """
+        CREATE TABLE IF NOT EXISTS SnapScheduleModuleKV (
+          key TEXT PRIMARY KEY,
+          value NOT NULL
+        ) WITHOUT ROWID;
+        INSERT OR IGNORE INTO SnapScheduleModuleKV (key, value) VALUES ('__snap_schedule_db_version', 1);
```

Comment:
If there **is** an older schema table existing then it won't be recreated with the new column. And now since the schema version table isn't found it will be created with version 2 and then the ALTER TABLE won't execute. So, the database system will not be in a consistent state.

Reply:
> If there **is** an older schema table existing then it won't be recreated with the new column. And now since the schema version table isn't found it will be created with version 2

Sure. But this method creates the schema with version 1, isn't it?

Reply:
> > If there **is** an older schema table existing then it won't be recreated with the new column. And now since the schema version table isn't found it will be created with version 2
> 
> Sure. But this method creates the schema with version 1, isn't it?

I was explaining how things would turn out if we implemented your earlier suggestion of starting with version 2 itself.

Reply:
Sure. What caught my attention was that the schema version was getting created with version 1 (if the table doesn't exist) and then we upgrade. However, if we create the schema with the highest (current) version then the upgrade isn't required.

Reply:
> Sure. What caught my attention was that the schema version was getting created with version 1 (if the table doesn't exist) and then we upgrade. However, if we create the schema with the highest (current) version then the upgrade isn't required.

Maybe I'm not getting it. But we still have to figure out a way of upgrading older schema. So, my idea is to create the DB with the old schema. Then create the schema version table with version 1. This holds good. Then check if the schema version is less than 2 and upgrade if so and bump up the schema version to 2. If there's another alteration required to schema we just add another test for version < 3 and add the alteration under that test condition and bump up the schema version to 3. This holds good even if a installation with old schema has skipped a version of upgrade and is upgrading to schema version 3 directly.
This is just like replaying the db upgrade journal :)
This doesn't need special casing the original db schema creation.

Reply:
Sure. I was just proposing another way and I'm not totally for or against one approach or the other :)

Another quick question - this is a new table that we introduce, isn't it? What about the version table that @batrick mentioned in standup a while ago?

Reply:
The table named `SnapScheduleModuleKV` is just an arbitrary key-value holder table.
The actual version would be stored in a record with the key named `__snap_schedule_db_version`.

This is what @batrick mentioned in standup a while age and pointed me another such implementation within Ceph from which I duplicated this code. fyi - he also mentioned that it wasn't worth the effort to refactor code to point to a single implementation as well.

Reply:
ACK. So SnapScheduleModuleKV already exists - just that this change introduces the (version) key.

Reply:
No the table doesn't already exist. `TABLE SnapScheduleModuleKV` will be introduced with this PR.